User story 18

yainporter opened the pull request:
Okay, I merged it with main before pushing, there should be no conflicts. I left in the error for User Story 14, until we decide what to do there.

yainporter committed:
Fix the testing issues for #application_pet_approved(id)

yainporter committed:
Merge branch 'main' of https://github.com/megroth2/adopt-dont-shop into user_story_16

yainporter committed:
Add in User Story 6 testing

yainporter committed:
Fix spacing issues for  show_spec describe blocks so they align correctly

yainporter committed:
Fix test for US17

yainporter committed:
Write extra tests for US 17

yainporter committed:
Fix admins/shelters/index.html.erb so that #self.pending_applications is not called

yainporter committed:
Change the pet/:id page to display age, breed, adoptable, and shelter name

yainporter committed:
Create set_adoptable_false method for Pet

yainporter committed:
Complete User Story 17

yainporter committed:
Write tests for User Story 18

yainporter committed:
Complete User Story 18, but have an error for User Story 14

yainporter committed:
Merge branch 'main' of https://github.com/megroth2/adopt-dont-shop into user_story_18

yainporter committed:
Merge with main

megroth2 commented on app/controllers/admins/applications_controller.rb:
```diff
@@ -2,7 +2,10 @@ class Admins::ApplicationsController < ApplicationController
-    if params[:filter].present?
+    if params[:filter] == "approved"
+      pet_application = Application.approve_or_deny_application_pet(params[:pet_id], params[:id], params[:filter])
+      Pet.find(params[:pet_id]).set_adoptable_false #is this chaining too much? I figured it's the same thing as: pet = Pet.find(params[:pet_id] -> pet.set_adoptable_false)
```
I don't think its too much!

megroth2 commented on app/models/application.rb:
```diff
@@ -41,11 +41,15 @@ def added_pets?
+      nil
+    else
+      false
+    end
```
good catch adding the nil back in

megroth2 commented on app/models/pet.rb:
```diff
@@ -9,11 +9,24 @@ def shelter_name
   end
+
+  def already_approved?
+    adoptable ? false : true
```
this is a mind bender lol. could we simplify it to just `!adoptable`? so it will return whatever the opposite of adoptable is?

megroth2 committed:
clarify the description of one of the tests so it reads a little smoother

megroth2 committed:
Remove save_and_open_page

megroth2 committed:
Remove commented out line from test

removed the already commented out line: `expect(@application_1.status).to eq("Pending")`
since this type of test would be included in a model spec, not a view spec

megroth2 commented on spec/models/pet_spec.rb:
```diff
@@ -44,7 +43,7 @@
 
   describe "instance methods" do
-    describe "#shelter_name" do
+    describe ".shelter_name" do
```
did you mean to change this back to `.`? I thought convention was to use `#`
